### PR TITLE
Bug 6819 - created common-vars.yml

### PR DESCRIPTION
**Pull request by balpurewal**
#### 📲 What

Create a common variables file and template which is replaced by stacks-cli

Also removed a few unused variables

#### 🤔 Why

Reduce the amount of repetitive values 
 
#### 🛠 How

Created common-vars.yml and common-vars-template.yml which are then referenced by all the pipelines.
The templated vars for infra and networking have been removed because those values come from the common-vars.yml

### CODE REVIEW

File: build/azDevOps/azure/air-infrastructure-data-vars.yml [hunk 1]
```diff
@@ -1,12 +1,4 @@
 variables:
-  - name: region
```
Comment (balpurewal) [inline build/azDevOps/azure/air-infrastructure-data-vars.yml:2]: these have been moved to common-vars.yml

File: build/azDevOps/azure/air-infrastructure-data-vars.yml [hunk 2]
```diff
@@ -74,35 +39,13 @@ variables:
     value: "$(self_repo_dir)/$(self_repo_tf_src)"
 
   # Versioning
-  - name: version_major
```
Comment (balpurewal) [inline build/azDevOps/azure/air-infrastructure-data-vars.yml:77]: these are already in version-data-vars.yml so can be removed

File: build/azDevOps/azure/air-infrastructure-data-vars.yml [hunk 3]
```diff
@@ -74,35 +39,13 @@ variables:
...
-    value: 1
-  - name: version_minor
-    value: 0
   - name: version_revision
     value: $[counter(join(variables['version_major'], join('-', variables['version_minor'])), 0)]
   - name: version_number
     value: $(version_major).$(version_minor).$(version_revision)
 
-
-  # Dashboard
```
Comment (balpurewal) [inline build/azDevOps/azure/air-infrastructure-data-vars.yml:87]: these variables aren't being used so have been removed

File: stackscli.yml [hunk 4]
```diff
@@ -4,10 +4,10 @@ pipeline:
       - name: build
         path: build/azDevOps/azure/templates/air-infrastructure-data.yml
       - name: variable
-        path: build/azDevOps/azure/air-infrastructure-data-vars.yml
+        path: build/common-vars.yml
```
Comment (balpurewal) [inline stackscli.yml:7]: the common-vars-template.yml is the only file that needs to be replaced by by stacks-cli

File: build/common-vars-template.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  - name: tf_state_rg
+    value: "{{ .Input.Terraform.Backend.Group }}"
+  - name: tf_state_storage
+    value: "{{ .Input.Terraform.Backend.Storage }}"
+  - name: tf_state_container
+    value: "{{ .Input.Terraform.Backend.Container }}"
+
+  # App Specific Taxonomy
+  - name: component
+    value: de  # core
```
Comment (adurkan-amido) [inline build/common-vars-template.yml:29]: In the Stacks CLI I believe there's a value `.Input.Business.Component` - I think it makes sense to use that here (and then move under variables)

**Review by adurkan-amido — COMMENTED** [review]
Very nice, much neater like this.

File: build/common-vars-template.yml (hunk 5 above)
Comment (balpurewal) [reply build/common-vars-template.yml:29]: sure

File: build/azDevOps/azure/air-infrastructure-data-vars.yml [hunk 6]
```diff
@@ -32,35 +21,11 @@ variables:
...
   - name: tf_state_key
-    value: $(domain)
+    value: core-data
   - name: TF_FILE_LOCATION
     value: $(self_repo_tf_src)
 
-  # App Specific Taxonomy
-  - name: component
-    value: de
-  - name: role
```
Comment (balpurewal) [inline build/azDevOps/azure/air-infrastructure-data-vars.yml:49]: this actually isn't being referenced anywhere so this has been removed

Comment (Danphillipz) [reply build/azDevOps/azure/air-infrastructure-data-vars.yml:49]: I raised this the other day to Andy, so there is a PR to bring this back in: https://github.com/Ensono/stacks-azure-data/pull/171/files

Docs also updated to reflect this being a value that is now used (added in anticipation)

Comment (satenderrathee) [reply build/azDevOps/azure/air-infrastructure-data-vars.yml:49]: Hey,I think Component is getting used in resource name

Comment (balpurewal) [reply build/azDevOps/azure/air-infrastructure-data-vars.yml:49]: Sorry wasn't clear 'role' isn't being used so is removed. 'component' has been moved to the common-variables.yml and stacks-cli will inject the component value to it

Comment (adurkan-amido) [reply build/azDevOps/azure/air-infrastructure-data-vars.yml:49]: @Danphillipz I will close that draft PR as this refactor includes the fix.

**Review by adurkan-amido — DISMISSED** [review]

**Review by satenderrathee — APPROVED** [review]

**Pull request merged**